### PR TITLE
Extract add_library_variant function

### PR DESCRIPTION
**Pull request by domin144**
The function now accepts named arguments. This will make the calls more readable, even when more arguments will be added.

### CODE REVIEW

**Comment by mplatings** [comment]
Thanks for this, I'm sympathetic to your desire to improve this.
However I think this is fixing the wrong problem. In my opinion the _right_ problem (that I recently created) is that there are too many arguments for each library variant. I'd like to reduce the number of arguments by unifying the multilib flags with the compile flags.
The benefit of keeping the library variants in a tabular format is that it's easy to spot inconsistencies.

Aside from the above, I believe you'll find that the multilib tests start failing if you do a clean build.

**Comment by domin144** [comment]
The next intended step was to add memory layout to the parameters list like this:
```
add_library_variant(
    aarch64
    COMPILE_FLAGS "-march=armv8-a"
    MULTILIB_FLAGS "--target=aarch64-none-unknown-elf"
    QEMU_PARAMS "-M virt -cpu cortex-a57"
    MEMORY_LAYOUT 0x40000000 4M 0x40400000 2M 8k
)
```
Do you have any suggestion how to proceed with this? Should I just add another column to the table?

An alternative I see, would be to replace QEMU_PARAMS and MEMORY_LAYOUT with some machine name. Maybe something like this:
```
set(virt_cortex-a57_qemu_params "-M virt -cpu cortex-a57")
set(virt_cortex-a57_memory_layout 0x40000000 4M 0x40400000 2M 8k)
# ... same for other machines

add_library_variants(
    add_library_variants(
    aarch64         ""                  "-march=armv8-a"                                        "--target=aarch64-none-unknown-elf" "virt_cortex-a57"
```

You are right about the multilib part. I forgot to remove the add_library_variants function and move `set(multilib_yaml_content "${multilib_yaml_content}" PARENT_SCOPE)` out of it. This should be easy to fix.

**Comment by mplatings** [comment]
I think it will be easier for me to judge if I can see your follow-up change. Can you make a separate pull request for that please?

File: cmake/to_meson_list.cmake [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# meson configuration file.
+# The delimiting brackes are not included.
+# Example output: "'foo', 'bar', 'baz'"
+
+function(to_meson_list input_list target_variable)
+    list(TRANSFORM input_list PREPEND "'")
+    list(TRANSFORM input_list APPEND "'")
+    list(JOIN input_list ", " input_list)
+    set(${target_variable} "${input_list}" PARENT_SCOPE)
+endfunction()
```
Comment (mplatings) [inline cmake/to_meson_list.cmake:11]: add terminating newline

**Review by mplatings — COMMENTED** [review]
Thanks for the other PR, that's given some very helpful context. If you can fix the multilib config then in principle I'm OK with this change.

File: cmake/to_meson_list.cmake [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Converts a cmake list to a string, which can be interpreted as list content in
+# meson configuration file.
+# The delimiting brackes are not included.
+# Example output: "'foo', 'bar', 'baz'"
+
+function(to_meson_list input_list target_variable)
+    list(TRANSFORM input_list PREPEND "'")
+    list(TRANSFORM input_list APPEND "'")
+    list(JOIN input_list ", " input_list)
+    set(${target_variable} "${input_list}" PARENT_SCOPE)
```
[the remark this answers is not in the document]
Comment (domin144) [reply cmake/to_meson_list.cmake]: Done

File: CMakeLists.txt [hunk 3]
```diff
@@ -860,66 +863,76 @@ function(get_compiler_rt_target_triple target_arch flags)
...
+        COMPONENT llvm-toolchain-libs
+    )
+endfunction()
 
-        string(APPEND multilib_yaml_content "  Flags:\n")
-        string(REPLACE " " ";" multilib_flags_list ${multilib_flags})
-        foreach(flag ${multilib_flags_list})
-          string(APPEND multilib_yaml_content "  - ${flag}\n")
-        endforeach()
+function(add_library_variants)
```
[the remark this answers is not in the document]
Comment (domin144) [reply CMakeLists.txt]: Done

File: cmake/to_meson_list.cmake [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Converts a cmake list to a string, which can be interpreted as list content in
+# meson configuration file.
+# The delimiting brackes are not included.
+# Example output: "'foo', 'bar', 'baz'"
+
+function(to_meson_list input_list target_variable)
```
[the remark this answers is not in the document]
Comment (domin144) [reply cmake/to_meson_list.cmake]: Done

**Review by mplatings — APPROVED** [review]

**Pull request merged**